tests/unit/test_util/test_urlUtils.py

```diff
@@ -0,0 +1,57 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2024 NV Access Limited, Noelia Ruiz Martínez, Leonard de Ruijter
+
+"""Unit tests for the urlUtils submodule."""
+
+import unittest
+from utils.urlUtils import isSamePageUrl
+
+
+class TestIsSamePageUrl(unittest.TestCase):
+	def test_samePage_basic(self):
+		self.assertTrue(isSamePageUrl("http://example.com/page#section", "http://example.com/page"))
+
+	def test_samePage_bothHaveFragments(self):
+		self.assertTrue(isSamePageUrl("http://example.com/page#section", "http://example.com/page#main"))
+
+	def test_differentPage(self):
+		self.assertFalse(isSamePageUrl("http://example.com/otherpage#section", "http://example.com/page"))
+
+	def test_noFragment(self):
+		"""URLs without fragments are not considered the same page."""
+		self.assertFalse(isSamePageUrl("http://example.com/page", "http://example.com/page"))
```

Comment:
why not? this seems like unexpected behaviour?

Reply:
I think that this test is valid, since the target and root url are equal and then the page maybe refreshed, but the link target is not an anchor for the same page.

Reply:
Perhaps the function should be renamed then `isLinkToSamePageAnchor` or similar?

Reply:
I really don't follow the logic on this one.
If I am on the page:

```
https://example.com/contact
```

And there is a standard menu on that page with a link called "Contact Us", linking to:

```
https://example.com/contact
```

I would expect that to say "Contact Us Link  in-page link", or whatever the label for a same page link is.
But this test is expected to fail, meaning that what I would anticipate is wrong here.
Can you explain this more?

Reply:
I agree. 
Either the concept is just for anchors, and we communicate it that way, or the concept is for same page links, which includes links to the exact same page.

Reply:
The concept is for same page links